agilent_infiniium_scope.py

```diff
@@ -409,67 +438,69 @@ def _parsePreamble(preamble):
-                ('yUnits',True),
-                ('maxBW',True),
-                ('minBW',True)]
-    preambleDict = {}
-    for key,val in zip(preambleKeys,preambleVals):
+    preamble_keys = [('byteFormat', True),
+                ('dataType', True),
+                ('numPoints', True),
+                ('count', True),
+                ('xStep', True),
+                ('xFirst', True),
+                ('xRef', True),
+                ('yStep', True),
+                ('yOrigin', True),
+                ('yRef', True),
+                ('coupling', True),
+                ('xRange', True),
+                ('xLeftDisplay', True),
+                ('yRange', True),
+                ('yCenterDisplay', True),
+                ('date', True),
+                ('time', True),
+                ('model', True),
+                ('acquisitionMode', True),
+                ('percentTimeBucketsComplete', True),
+                ('xUnits', True),
+                ('yUnits', True),
+                ('maxBW', True),
+                ('minBW', True)]
+    preample_dict = {}
+    for key, val in zip(preamble_keys, preamble_vals):
         if key[1]:
-            preambleDict[key[0]] = val
-    def unitType(num):
-        if num=='1':
+            preample_dict[key[0]] = val
+
+    def unit_type(num):
+        if num == '1':
             return 'V'
-        elif num=='2':
+        elif num == '2':
             return 'ns'
         else:
             raise Exception('Units not time or voltage')
-    preambleDict['xUnit'] = unitType(preambleDict['xUnits'])
-    preambleDict['yUnit'] = unitType(preambleDict['yUnits'])
-    return (preambleDict)
+    preample_dict['xUnit'] = unit_type(preample_dict['xUnits'])
+    preample_dict['yUnit'] = unit_type(preample_dict['yUnits'])
+    return (preample_dict)
+
 
-def _parseBinaryData(data, wordLength):
+def _parseBinaryData(data, word_length):
     """Parse binary data packed as string of RIBinary
     """
-    formatChars = {'1':'b','2':'h', '4':'f'}
-    formatChar = formatChars[str(wordLength)]
-
-    #Get rid of header crap
-    #unpack binary data
-    if wordLength == 1:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = numpy.array(unpack(formatChar*(len(dat)/wordLength),dat))
-    elif wordLength == 2:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = dat[-calcsize('>' + formatChar*(len(dat)/wordLength)):]
-        dat = numpy.array(unpack('>' + formatChar*(len(dat)/wordLength),dat))
-    elif wordLength == 4:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = dat[-calcsize('>' + formatChar*(len(dat)/wordLength)):]
-        dat = numpy.array(unpack('>' + formatChar*(len(dat)/wordLength),dat))      
+    format_chars = {'1': 'b', '2': 'h', '4': 'f'}
+    format_char = format_chars[str(word_length)]
+
+    # Get rid of header
+    # unpack binary data
+    if word_length == 1:
```

Comment:
Good point about the data. In the first error, you might want to include just the first byte. In the second, probably just including the length is enough.

`//` _is_ a real thing. It means do integer division, as opposed to floating point division. In python 2 if you divide two integers with `/` you always get an int, but in python 3 dividing `/` gives a float and you have to use `//` to get integer division. You can turn on the python 3 behavior by doing `from __future__ import division`, which we've gotten in the habit of doing for new code. But in any case, `//` does integer division with or without that import, so it's a good habit to get into :-)